include/ensmallen_bits/cmaes/cmaes_impl.hpp

```diff
@@ -313,6 +325,89 @@ typename MatType::elem_type CMAES<SelectionPolicyType>::Optimize(
   return overallObjective;
 }
 
+// Transforms the candidate into the given bounds.
+template<typename SelectionPolicyType>
+template<typename ElemType, typename BaseMatType>
```

Comment:
Instead of using another template parameter here, we used:

```c+=
// Convenience typedef.
typedef typename BaseMatType::elem_type ElemType;
```

in the past.

Reply:
But if we don't use `ElemType` as the template parameter here won't it give `not declared in this scope` error. Please correct me if I am wrong.

Reply:
If you put:

```c++
// Convenience typedef.
typedef typename BaseMatType::elem_type ElemType;
```

right at the top it should work, unless I missed something.